modules/account-lib/test/unit/coin/sol/transactionBuilder/stakingActivateBuilder.ts

```diff
@@ -0,0 +1,279 @@
+    });
+
+    it('build a staking activate tx with the same sender and staking address', async () => {
+      const txBuilder = stakingBuilder();
+      txBuilder.sender(wallet.pub);
+      txBuilder.stakingAddress(wallet.pub);
+      txBuilder.validator(validator.pub);
+      txBuilder.amount(amount);
+      txBuilder.sign({ key: wrongAccount.prv });
+      await txBuilder.build().should.rejectedWith('Sender address cannot be the same as the Staking address');
+    });
+
+    it('build when nonce is not provided', async () => {
+      const txBuilder = factory.getStakingActivateBuilder();
+      txBuilder.sender(wallet.pub);
+      txBuilder.stakingAddress(stakeAccount.pub);
+      txBuilder.amount(amount);
+      txBuilder.validator(validator.pub);
+      txBuilder.sign({ key: wallet.prv });
+      await txBuilder.build().should.rejectedWith('Invalid transaction: missing nonce blockhash');
+    });
+
+    it('build when sender is not provided', async () => {
+      const txBuilder = factory.getStakingActivateBuilder();
+      txBuilder.stakingAddress(stakeAccount.pub);
+      txBuilder.amount(amount);
+      txBuilder.validator(validator.pub);
+      txBuilder.nonce(recentBlockHash);
+      txBuilder.sign({ key: wallet.prv });
+      await txBuilder.build().should.rejectedWith('Invalid transaction: missing sender');
+    });
+
+    it('build when stakingAddress is not provided', async () => {
+      const txBuilder = factory.getStakingActivateBuilder();
+      txBuilder.sender(wallet.pub);
+      txBuilder.amount(amount);
+      txBuilder.validator(validator.pub);
+      txBuilder.nonce(recentBlockHash);
+      txBuilder.sign({ key: wallet.prv });
+      await txBuilder.build().should.rejectedWith('Staking Address must be set before building the transaction');
+    });
+
+    it('build when validator is not provided', async () => {
+      const txBuilder = factory.getStakingActivateBuilder();
+      txBuilder.stakingAddress(stakeAccount.pub);
+      txBuilder.sender(wallet.pub);
+      txBuilder.amount(amount);
+      txBuilder.nonce(recentBlockHash);
+      txBuilder.sign({ key: wallet.prv });
+      await txBuilder.build().should.rejectedWith('Validator must be set before building the transaction');
+    });
+
+    it('to sign twice with the same key', () => {
+      const txBuilder = factory.from(testData.STAKING_ACTIVATE_UNSIGNED_TX);
+      txBuilder.sign({ key: wallet.prv });
+      should(() => txBuilder.sign({ key: wallet.prv })).throwError('Duplicated signer: ' + wallet.prv?.toString());
+    });
+  });
+
+  describe('From and sign', () => {
+    describe('Succeed', () => {
+      it('build from an unsigned staking activate and sign it', async () => {
+        const txBuilder = factory.from(testData.STAKING_ACTIVATE_UNSIGNED_TX);
+
+        txBuilder.sign({ key: wallet.prv });
+        txBuilder.sign({ key: stakeAccount.prv });
```

Comment:
before you sign, could you verify the following:
- build tx and verify `unsignedTx.id === 'UNAVAILABLE'`
- set fee in builder to `5000` and verify output fee from `explainTransaction` is 10000 since this should require 2 sigs

Reply:
or maybe, we should have a separate set of tests for `explainTransaction`

Reply:
We could create a ticket to add tests for explainTransaction once Activate,Deactivate and Withdraw builder get done.

We can add those test here https://github.com/BitGo/BitGoJS/blob/master/modules/account-lib/test/unit/coin/sol/transaction.ts#L217

Do you agree?

Reply:
SGTM